csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -121,16 +125,35 @@ public sealed class CodedInputStream : IDisposable
         /// <summary>
         /// Creates a new CodedInputStream reading data from the given byte array.
         /// </summary>
-        public CodedInputStream(byte[] buffer) : this(null, ProtoPreconditions.CheckNotNull(buffer, "buffer"), 0, buffer.Length, true)
-        {            
+        [SecuritySafeCritical]

```

Comment:
why is adding these attributes important? Are they necessary for some reason or they are just nice-to-have?

Reply:
Definitely required as described in a comment below